src/decompose.rs

```diff
@@ -93,21 +103,28 @@ impl<I: Iterator<Item=char>> Iterator for Decompositions<I> {
 
     #[inline]
     fn next(&mut self) -> Option<char> {
-        while self.ready == 0 && !self.done {
+        while self.ready.end == 0 {
             match (self.iter.next(), &self.kind) {
```

Comment:
Ah, that's what that `done` field was doing! Let me see if I can restructure in a way that addresses both this and your other comment on the function.

Reply:
Yeah, I'm not sure exactly what resumption means in this case.  Does it, for example, restart the iterator at its beginning?  Or is it for streaming iterators that want to return `None` and then make subsequent values available?  

In either case, our loop here would get really confused by a `Some` following a `None`: we'd 1) sort the characters we had left on receiving `None`, 2) potentially emit some of them, and then 3) push some more decomposed characters on from the subsequent `Some`.  We'd then potentially emit non-starters out of combining class order due to the premature sort at step 1.

FWIW, the existence of the [`Fuse`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fuse) combinator signals to me that it's not _really_ expected to call `next` on an exhausted iterator.  This combinator takes an arbitrary iterator and, at some runtime cost, ensures it always returns `None` after returning `None` the first time.

The state of the docs here is a little disappointing.  The `Future`s ones do a better job: https://doc.rust-lang.org/std/future/trait.Future.html#panics.

Reply:
Ah, cool, `fuse` actually seems like exactly what we want here so that there's at least an opportunity for specialization to eliminate the cost of the `done` check.